tardis/tests/tests_slow/test_w7.py

```diff
@@ -39,18 +37,75 @@ class TestW7(object):
         * last_line_interaction_shell_id |     * montecarlo_nu
         * nubar_estimators               |     * last_line_interaction_angstrom
         * ws                             |     * j_blues_norm_factor
+
+    * spectrum.npz
+    Contents (all (.npy)):
+        * luminosity_density_nu          | * wavelength
+        * delta_frequency                | * luminosity_density_lambda
+    """
+
+    # TODO: make this fixture ingest data from an HDF5 file.
+    datadir_path = os.path.join(os.path.expanduser(os.path.expandvars(
+            request.config.getvalue('slow-test-data'))), "w7")
+
+    ndarrays = dict(np.load(os.path.join(datadir_path, "ndarrays.npz")))
+    quantities = dict(np.load(os.path.join(datadir_path, "quantities.npz")))
+    spectrum = dict(np.load(os.path.join(datadir_path, "spectrum.npz")))
+
+    # Associate CGS units to ndarrays of baseline quantities.
+    quantities['j_blues_norm_factor'] = (
+        u.Quantity(quantities['j_blues_norm_factor'], '1 / (cm2 s)'))
+
+    quantities['last_line_interaction_angstrom'] = (
+        u.Quantity(quantities['last_line_interaction_angstrom'], 'Angstrom'))
+
+    quantities['luminosity_inner'] = (
+        u.Quantity(quantities['luminosity_inner'], 'erg / s'))
+
+    quantities['montecarlo_luminosity'] = (
+        u.Quantity(quantities['montecarlo_luminosity'], 'erg / s'))
+
+    quantities['montecarlo_virtual_luminosity'] = (
+        u.Quantity(quantities['montecarlo_virtual_luminosity'], 'erg / s'))
+
+    quantities['montecarlo_nu'] = u.Quantity(quantities['montecarlo_nu'], 'Hz')
+    quantities['t_rads'] = u.Quantity(quantities['t_rads'], 'K')
+
+    # Associate CGS units to ndarrays of baseline quantities (spectrum).
+    spectrum['luminosity_density_nu'] = (
+        u.Quantity(spectrum['luminosity_density_nu'], 'erg'))
+
+    spectrum['delta_frequency'] = u.Quantity(spectrum['delta_frequency'], 'Hz')
+    spectrum['wavelength'] = u.Quantity(spectrum['wavelength'], 'Angstrom')
+
+    spectrum['luminosity_density_lambda'] = (
+        u.Quantity(spectrum['luminosity_density_lambda'], 'erg / (Angstrom s)'))
+
+    ndarrays.update(quantities, **spectrum)
+    return ndarrays
+
+
+@pytest.mark.skipif(not pytest.config.getoption("--slow"),
+                    reason="slow tests can only be run using --slow")
+@pytest.mark.skipif(not pytest.config.getvalue("slow-test-data"),
+                    reason="--slow-test-data was not specified")
+@pytest.mark.skipif(not pytest.config.getvalue("atomic-dataset"),
+                    reason="--atomic-dataset was not specified")
```

Comment:
I suggest you create a mark that combines these.
In the end it should look like this:
```
@runslow
class TestW7(object):
```